composer/utils/checkpoint.py

```diff
@@ -532,32 +531,55 @@ def save_checkpoint(
 
     # Sharded checkpoints get their own little folder.
     if state.fsdp_sharded_state_dict_enabled:
+        if using_torch_2():
+            # Make state dict.
+            if weights_only:
```

Comment:
check whether we can _only_ bump optimizers to the top level and otherwise not change this logic from before this pr

Reply:
ok changed it to just bump optimizers